eppo_client/client.py

```diff
@@ -292,14 +300,21 @@ def get_bandit_action_detail(
         self,
         flag_key: str,
         subject_key: str,
-        subject_context: Attributes,
-        actions: ActionContexts,
+        subject_context: Union[Attributes, AttributesDict],
+        actions: Union[ActionContexts, ActionContextsDict],
         default: str,
     ) -> BanditResult:
+        subject_attributes = convert_subject_context_to_attributes(subject_context)
+        action_contexts = convert_actions_to_action_contexts(actions)
+
         # get experiment assignment
         # ignoring type because Dict[str, str] satisfies Dict[str, str | ...] but mypy does not understand
         variation = self.get_string_assignment(
-            flag_key, subject_key, subject_context.categorical_attributes, default  # type: ignore
+            flag_key,
+            subject_key,
+            subject_attributes.categorical_attributes
+            | subject_attributes.numeric_attributes,  # type: ignore
```

Comment:
minor: one thing that still bothers me is that:
```py
subject_attributes.categorical_attributes | subject_attributes.numeric_attributes != subject_context
```
It's just seems very hacky and very easy to shoot oneself in the foot with this, as it may introduce subtle evaluation differences. (e.g., `get_string_assignment()` returns a different value from here when called by the user).

As one example, attributes that are not one of `[str,int,float,bool]` are silently dropped for bandits. (In normal flag evaluation, they are convertible to string.)

It would help to either cover this with a lot of tests. Or maybe refactor the code so that  we perform similar transformation on all flag evaluation before attributes enter rules evaluation. (So that flag evaluation engine only deals with floats and strings.)

Reply:
Thanks, this is a great comment! 

> As one example, attributes that are not one of [str,int,float,bool] are silently dropped for bandits

But the attribute type covers all the accepted cases: `AttributeType = Union[str, int, float, bool, None]`; Yes `None` gets dropped but it indicates absence anyway and would evaluate the same way whether we set the value to None or remove the key, value pair altogether.

Perhaps the problem is that we are not dropping none [str,int,float,bool] values when evaluating `get_string_assignment`?

Reply:
As another inconsistency: `int`s may loose precision.

Reply:
Not so worried about losing precision -- the loss in precision is orders of magnitude less than the statistical uncertainty in the bandit model